modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -609,27 +600,20 @@
   ;; automatically retry the query if it times out or otherwise fails. This is on top of the auto-retry added by
   ;; `execute`
   (let [thunk (fn []
-                (post-process-native respond
-                                     (execute-bigquery-on-db
-                                      database
-                                      sql
-                                      parameters
-                                      cancel-chan)
-                                     cancel-chan))]
+                (execute-bigquery
+                 respond
+                 (:details database)
+                 sql
+                 parameters
+                 cancel-chan))]
     (try
       (thunk)
       (catch Throwable e
         (let [ex-data (u/all-ex-data e)]
-          (if (and (not (::cancelled? ex-data))
-                   (or (:retryable? ex-data) (not (qp.error-type/client-error? (:type ex-data)))))
+          (if (:retryable? ex-data)
```

Comment:
This would retry random exceptions and cause loops, we only care to retry if bigquery says we should